driver-scala/src/test/scala/org/mongodb/scala/ScalaPackageSpec.scala

```diff
@@ -141,4 +140,33 @@ class ScalaPackageSpec extends BaseSpec {
     val javaCredential5 = JMongoCredential.createGSSAPICredential("userName")
     scalaCredential5 should equal(javaCredential5)
   }
+
+  it should "implicitly convert to org.bson.document with type fidelity" in {
+
+    val bsonDocument = Document(
+      "null" -> BsonNull(),
+      "int32" -> BsonInt32(32),
+      "int64" -> BsonInt64(64L),
```

Comment:
I'd change `64L` -> `Long.MaxValue` for reproducing issue in JIRA ticket.

FYI, I am OP of the bug

Reply:
Sounds good